### PR TITLE
Support Zstd compression for gRPC

### LINKED ISSUE #5091: support zstd compressor for grpc
**Is your feature request related to a problem? Please describe.**
Inter AZ network costs between distributor and ingesters are a major issue faced by all. Currently one way to bring the network costs down is to enable gzip compression on grpc calls between distributors and ingesters

**Describe the solution you'd like**
with a pure go implementation of [zstd](https://github.com/klauspost/compress/tree/master/zstd#zstd) now available, we can look at trying out zstd compression as an alternative to gzip.

### PR DESCRIPTION
**Pull request by ssesha**
**What this PR does**:
Add zstd compression for grpc
Below output with zstd level BetterCompression as [here](https://github.com/klauspost/compress/blob/master/zstd/encoder_options.go#L162)

Screenshot 2023-01-11 at 3 08 20 PM

We see 

**Which issue(s) this PR fixes**:
Fixes #5091 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by alanprot — DISMISSED** [review]

**Comment by alanprot** [comment]
This looks good!

Can you update the changelog?

**Comment by ssesha** [comment]
Thanks @alanprot . Have updated the changelog

**Review by alanprot — APPROVED** [review]

**Comment by alanprot** [comment]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * [ENHANCEMENT] Query Frontend: Log Vertical sharding information when `query_stats_enabled` is enabled. #5037
 * [ENHANCEMENT] Ingester: The metadata APIs should honour `querier.query-ingesters-within` when `querier.query-store-for-labels-enabled` is true. #5027
 * [ENHANCEMENT] Query Frontend: Skip instant query roundtripper if sharding is not applicable. #5062
+* [ENHANCEMENT] Added zstd as an option for grpc compression #5092
```
Comment (yeya24) [inline CHANGELOG.md:14]: Let's call it `FEATURE` instead?

**Review by yeya24 — APPROVED** [review]
Thanks for the contribution!

File: CHANGELOG.md (hunk 1 above)
Comment (ssesha) [reply CHANGELOG.md]: Done

**Review by yeya24 — APPROVED** [review]

**Pull request merged**